src/components/atoms/VideoOverlayButton/VideoOverlayButton.tsx

```diff
@@ -27,35 +41,18 @@ export interface VideoOverlayButtonProps extends ContainerClassName {
 }
 
 export const VideoOverlayButton: React.FC<VideoOverlayButtonProps> = ({
-  variant,
+  variant = "microphone",
   defaultValue,
   onEnabledChanged,
   containerClassName,
 }) => {
-  const [enabled, toggle] = useToggle(defaultValue);
+  const [state, toggle] = useToggle(defaultValue);
 
   useEffect(() => {
-    onEnabledChanged(enabled);
-  }, [enabled, onEnabledChanged]);
+    onEnabledChanged(state);
+  }, [state, onEnabledChanged]);
 
-  let iconEnabled: IconDefinition = faMicrophone;
-  let iconDisabled: IconDefinition = faMicrophoneSlash;
-  switch (variant) {
-    case "microphone":
-      iconEnabled = faMicrophone;
-      iconDisabled = faMicrophoneSlash;
-      break;
-    case "camera":
-      iconEnabled = faVideo;
-      iconDisabled = faVideoSlash;
-      break;
-    case "audio":
-      iconEnabled = faVolumeUp;
-      iconDisabled = faVolumeMute;
-      break;
-    default:
-      assertUnreachable(variant);
-  }
+  const { enabled, disabled } = ICONS[variant];
```

Comment:
One line for the entire `switch` with type safety preserved, declarative code at the top mapping the buttons and no need for assertions